### ISSUE TITLE
Content-Type with charset

### ISSUE THREAD

**Issue by skywall** [System-Glitch/goyave#101]
## Proposal

Current implementation automatically Unmarshal request data only in a situation when header `Content-Type` equals to `application/json`. [RFC 7231](https://tools.ietf.org/html/rfc7231) describing `Hypertext Transfer Protocol (HTTP/1.1): Semantics and Content` specifies that `Content-Type` should contain media-type what is defined not only by data format (`application/json`) but also by additional metadata about processing (`charset`).

I'm not that experienced in Go, but I guess the problem is based on this [check](https://github.com/System-Glitch/goyave/blob/master/middleware.go#L75).

Some of the heavily used serialization libraries on Android – kotlinx.serialization, Moshi – put charset into `Content-Type` by default and it might look similar to this one: `Content-Type: application/json; charset=utf-8`

To summarize, with this curl request, content of `request.Data` is unmarshalled automatically:
```
curl -X "POST" "http://localhost:8080/v1/login" \
     -H 'Content-Type: application/json' \
     -d $'{
  "email": "my@email.com",
  "password": "12345678"
}'
```

Whereas with this request not:
```
curl -X "POST" "http://localhost:8080/v1/login" \
     -H 'Content-Type: application/json; charset=utf-8' \
     -d $'{
  "email": "my@email.com",
  "password": "12345678"
}'
```

### Possible drawbacks

I'm not sure if it's sufficient just change mentioned check from `==` (equals) into `contains()` or it is necessary to take provided charset into account when data are unmarshalled.

**Comment by System-Glitch** [comment]
Hello, thank you very much. This can be considered a bug.

**Comment by System-Glitch** [comment]
@skywall A new version (2.10.2) has been published. I recommend you to upgrade.

**Comment by skywall** [comment]
Works great! Thank you very much.
